### PR TITLE
Cleanup testing reference manual section

### PR DESCRIPTION
**Pull request by kprosise**
Applied style guide recommendations and edited to increase readability.

QA Steps: linter and spellcheck used in editor. Checked rendered output for correctness. No issues found.

This commit addresses FFTK-2796
This commit applies to FFTK-1702

# PR Template and Checklist

Please complete as much as possible to speed up the reviewing process.

Readiness and adding reviewers as appropriate is required.

All PRs should be reviewed by a technical writer/documentation team and a peer.
If effecting customers—which is a majority of content changes—a member of Customer Success must also review.

## Readiness

* [x] Merge (pending reviews)
* [ ] Merge after _date or event_
* [ ] Draft

## Overview

In addition to cleanup, also added an overview summary to the title/toc page for the section.

## Checklist

* [x] Run spelling and grammar check, preferably with linter.
* [x] Avoid changing any header associated with a link/reference.
* [ ] Step through instructions (or ask someone to do so).
* [x] Review for [wordiness](https://languagetool.org/insights/post/wordiness/)
* [x] Match tone and style of page/section.
* [ ] Run `make linkcheck`.
* [x] View HTML in a browser to check rendering.
* [x] Use [semantic newlines](https://bobheadxi.dev/semantic-line-breaks/).
* [x] follow best practices for commits.
  * [x] Descriptive title written in the imperative.
  * [x] Include brief overview of QA steps taken.
  * [x] Mention any related issues numbers.
  * [x] End message with sign off/DCO line (`-s, --signoff`).
  * [x] Sign commit with your gpg key (`-S, --gpg-sign`).
  * [x] Squash commits if needed.
* [x] Request PR review by a technical writer and at least one peer.

## Comments

_Any thing else that a maintainer/reviewer should know._
_This could include potential issues, rational for approach, etc._

### CODE REVIEW

**Comment by doanac** [comment]
Docs for 9729a522135a82521c769d67e037cf0eab66efe3 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2592/docs/artifacts/html/index.html

File: source/reference-manual/testing/lmp-testplan.rst [hunk 1]
```diff
@@ -187,80 +183,61 @@ There are 2 scenarios for boot testing:
...
-all testing factories pull latest source from ``lmp-manifest`` project
-and merge to their working branches.
-Successful build in the testing factory triggers tests on the devices.
+If the build is successful, all testing Factories pull the latest source from ``lmp-manifest`` and merge to their working branches.
+A successful build in the testing Factory triggers tests on the devices.
 OTA update is delivered to the *rolling update* devices.
-This also triggers a testing round on the new target.
-For a release candidate build additional manual tests are performed.
+This also triggers a testing round on the new Target.
+For a release candidate buil,d additional manual tests are performed.
```
Comment (mwasilew) [inline source/reference-manual/testing/lmp-testplan.rst:240]: should be ``build, ``

**Review by mwasilew — CHANGES_REQUESTED** [review]
I found 1 typo. Apart from that LGTM. Many thanks for doing that!

File: source/reference-manual/testing/lmp-testplan.rst (hunk 1 above)
Comment (kprosise) [inline source/reference-manual/testing/lmp-testplan.rst:240]:
```suggestion
For a release candidate build, additional manual tests are performed.
```

Comment (kprosise) [reply source/reference-manual/testing/lmp-testplan.rst:240]: weird, I recall seeing that while editing, and thought I had fixed it! Thanks for spotting it!

**Review by kprosise — COMMENTED** [review]
address feedback

**Comment by doanac** [comment]
Docs for afe6d8df58591e931b7ff9f1899e1e0004ce7641 are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2593/docs/artifacts/html/index.html

**Review by mwasilew — APPROVED** [review]

**Comment by doanac** [comment]
Docs for a547999e15f1effa8201addcbb932d4f7634463d are browsable at: https://ci.foundries.io/projects/fio-docs/builds/2596/docs/artifacts/html/index.html

**Pull request merged**